app/services/candidate_interface/section.rb

```diff
@@ -18,5 +23,62 @@ def science_gcse?(policy)
           .select(&:science_gcse_needed?)
           .all?(&:unsubmitted?)
     end
+
+    def maths_gcse?(_policy)
+      subject = params[:subject]
+
+      subject && subject == 'maths'
+    end
+
+    def english_gcse?(_policy)
+      subject = params[:subject]
+
+      subject && subject == 'english'
+    end
+
+    def eql?(other)
+      id == other.id
+    end
+
+    def self.all
+      [
+        Section.new(:personal_details, controller: 'CandidateInterface::PersonalDetails'),
+        Section.new(:contact_details, controller: 'CandidateInterface::ContactDetails'),
+        Section.new(:training_with_a_disability, controller: 'CandidateInterface::TrainingWithADisability'),
+        Section.new(:interview_preferences, controller: 'CandidateInterface::InterviewAvailability'),
+        Section.new(:equality_and_diversity, controller: 'CandidateInterface::EqualityAndDiversity'),
+        Section.new(:becoming_a_teacher, controller: 'CandidateInterface::PersonalStatement'),
+        Section.new(
+          :science_gcse,
+          controller: 'CandidateInterface::Gcse',
+          editable_condition: ->(section, policy) { section.science_gcse?(policy) },
```

Comment:
I made the suggestion to pass a block here but the way it's been implemented means it is unnecessary.

The point was to avoid defining `*_gcse?` methods on the `Section` class because that method only applies to a GCSE section with subject *. If another GCSE was added (unlikely of course) then a new method would need to be added.

I expected the whole condition (the contents of the `*_gcse?` method) to be passed directly to the block.

```ruby
      Section.new(
        :english_gcse,
        controller: 'CandidateInterface::Gcse',
        editable_condition: lambda { |policy|
          params = policy.params
          subject = params[:subject]

          (subject && subject == 'english') || policy.controller_path.include?('candidate_interface/gcse/english')
        },
      ),
```

Reply:
Also, how do you feel about `editable_if:` instead of `editable_condition:`